crates/ruff_workspace/src/configuration.rs

```diff
@@ -322,26 +296,14 @@ impl Configuration {
     }
 
     pub fn from_options(options: Options, project_root: &Path) -> Result<Self> {
+        // TODO warn about legacy options
+        let lint = if let Some(lint) = options.lint {
```

Comment:
You could do this with option chaining, probably, if you want :)

Reply:
Not sure how, the following doesn't work because `combine` consumes `options.lint_top_level`:

```rust
let lint = options.lint.map_or_else(|| options.lint_top_level, |lint| lint.combine(options.lint_top_level))
```